cpp/src/neighbors/detail/nn_descent.cuh

```diff
@@ -829,36 +830,29 @@ void GnndGraph<Index_t>::sample_graph_new(InternalID_t<Index_t>* new_neighbors,
 template <typename Index_t>
 void GnndGraph<Index_t>::init_random_graph()
```

Comment:
Could you add a docstring to describe what properties the random graph is expected to have? What is ensured by the current PR to fix the low recall issue? E.g. is it guaranteed to be fully connected?